packages/core/utils/lib/async.js

```diff
@@ -0,0 +1,55 @@
+'use strict';
+
+const { chunk } = require('lodash/fp');
+
+function pipeAsync(...methods) {
+  return async (data) => {
+    let res = data;
+
+    for (const method of methods) {
+      res = await method(res);
+    }
+
+    return res;
+  };
+}
+
+/**
+ * @type { import('./async').mapAsync }
+ */
+function mapAsync(promiseArray, { concurrency = Infinity } = {}) {
+  const appliedConcurrency = concurrency > promiseArray.length ? promiseArray.length : concurrency;
+  const promiseArrayChunks = chunk(appliedConcurrency)(promiseArray);
+
+  return async (iteratee) => {
+    return promiseArrayChunks.reduce(async (prevChunksPromise, chunk, chunkIndex) => {
+      // Need to await previous promise in order to respect the concurrency option
+      const prevChunks = await prevChunksPromise;
+      // As chunks can contain promises, we need to await the chunk
+      const awaitedChunk = await Promise.all(chunk);
+      const transformedPromiseChunk = await Promise.all(
+        // Calculating the index based on the original array, we do not want to have the index of the element inside the chunk
+        awaitedChunk.map((value, index) => iteratee(value, chunkIndex * appliedConcurrency + index))
+      );
+
+      return prevChunks.concat(transformedPromiseChunk);
+    }, Promise.resolve([]));
+  };
+}
+
+/**
+ * @type { import('./async').reduceAsync }
+ */
+function reduceAsync(promiseArray) {
+  return (iteratee, initialValue) =>
+    promiseArray.reduce(async (previousPromise, currentValue, index) => {
+      const previousValue = await previousPromise;
+      return iteratee(previousValue, await currentValue, index);
+    }, Promise.resolve(initialValue));
+}
```

Comment:
Using curry, it would be:

```js
const reduceAsync = curry((iteratee, initialValue, array) => { ... });
```

Reply:
I don't want to use the curry function, it seems to me to obfuscate the function and make it even more difficult to read.
What is the added value to use curry instead of splitting the function in the code?

Reply:
It's to be similar to the other utils we use from `lodash/fp`.
```fp
const { map } = require('lodash/fp');

const result = map(x => x, [1, 2, 3])
// is the same as
const result = map(x => x)([1, 2, 3]) 
// which is the same as
const mapFunc = map(x => x);
const result = mapFunc([1, 2, 3])
```

It can be really handy. For instance, the dev can choose to use an intermediate `mapFunc` function if it's used several times or not if it's used only once.